src/python_testing/TC_AVSMTestBase.py

```diff
@@ -0,0 +1,173 @@
+        asserts.assert_equal(vdoSupport, cluster.Bitmaps.Feature.kVideo, "Video Feature is not supported.")
+
+        # Check if video stream has already been allocated
+        aAllocatedVideoStreams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedVideoStreams
+        )
+        logger.info(f"Rx'd AllocatedVideoStreams: {aAllocatedVideoStreams}")
+        if len(aAllocatedVideoStreams) > 0:
+            return
+
+        # Allocate one for the test steps
+        aRankedStreamPriorities = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.RankedVideoStreamPrioritiesList
+        )
+        logger.info(f"Rx'd RankedVideoStreamPrioritiesList: {aRankedStreamPriorities}")
+        aRateDistortionTradeOffPoints = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.RateDistortionTradeOffPoints
+        )
+        logger.info(f"Rx'd RateDistortionTradeOffPoints: {aRateDistortionTradeOffPoints}")
+        aMinViewport = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.MinViewport
+        )
+        logger.info(f"Rx'd MinViewport: {aMinViewport}")
+        aVideoSensorParams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.VideoSensorParams
+        )
+        logger.info(f"Rx'd VideoSensorParams: {aVideoSensorParams}")
+        aMaxEncodedPixelRate = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.MaxEncodedPixelRate
+        )
+        logger.info(f"Rx'd MaxEncodedPixelRate: {aMaxEncodedPixelRate}")
+
+        try:
+            asserts.assert_greater(len(aRankedStreamPriorities), 0, "RankedVideoStreamPrioritiesList is empty")
+            asserts.assert_greater(len(aRateDistortionTradeOffPoints), 0, "RateDistortionTradeOffPoints is empty")
+            videoStreamAllocateCmd = commands.VideoStreamAllocate(
+                streamUsage=aRankedStreamPriorities[0],
+                videoCodec=aRateDistortionTradeOffPoints[0].codec,
+                minFrameRate=aVideoSensorParams.maxFPS,
+                maxFrameRate=aVideoSensorParams.maxFPS,
+                minResolution=aRateDistortionTradeOffPoints[0].resolution,
+                maxResolution=aRateDistortionTradeOffPoints[0].resolution,
+                minBitRate=aRateDistortionTradeOffPoints[0].minBitRate,
+                maxBitRate=aRateDistortionTradeOffPoints[0].minBitRate,
+                minFragmentLen=4000,
+                maxFragmentLen=4000,
+            )
+            await self.send_single_cmd(endpoint=endpoint, cmd=videoStreamAllocateCmd)
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
```

Comment:
Expected response is VideoStreamAllocateResponse with a StreamID.

Reply:
Okay, Thank you for update. We had changed it to streamId